lib/search.js

```diff
@@ -81,18 +100,15 @@ module.exports = new class LibSearch {
             return ret.join('')
         }
     }
-
     optimizeSubs(response, input) {
         // based on OpenSRTJS, under MIT - Copyright (c) 2014 Eóin Martin
 
         let fileTags
         let fileTagsDic = Object()
         const normalize = this.normalizeProt()
         const matchTags = (sub, maxScore) => {
-            if (!input.filename) return 0
 
-            if (!fileTags) fileTags = normalize(input.filename).toLowerCase().match(/[a-z0-9]{2,}/gi)
-            if (!fileTags.length) return 0
+            if (!input.tags || !input.tags.length) return 0

```

Comment:
you're destroying the function by not keeping fileTags up to date: it's always gonna be undefined now, yet it is still used a few lines below